pkg/controller/certificates/controller.go

```diff
@@ -181,6 +188,47 @@ func (cc *CertificateController) maybeSignCertificate(key string) error {
 	}
 	csr := obj.(*certificates.CertificateSigningRequest)
 
+	func() {
+		if cc.approveAllKubeletCSRsForGroup != "" ||
+			IsCertificateRequestApproved(csr) ||
+			IsCertificateRequestDenied(csr) {
+			return
+		}
+
+		isKubeletBootstrapGroup := false
+		for _, g := range csr.Spec.Groups {
+			if g == cc.approveAllKubeletCSRsForGroup {
+				isKubeletBootstrapGroup = true
+				break
+			}
+		}
+		if !isKubeletBootstrapGroup {
+			return
+		}
+
+		x509cr, err := utilcertificates.ParseCertificateRequestObject(csr)
+		if err != nil {
+			glog.Errorf("unable to parse csr %q: %v", csr.ObjectMeta.Name, err)
+			return
+		}
+		if !reflect.DeepEqual([]string{"system:nodes"}, x509cr.Subject.Organization) {
+			return
+		}
+		if !strings.HasPrefix(x509cr.Subject.CommonName, "system:node:") {
+			return
+		}
+		if len(x509cr.DNSNames)+len(x509cr.EmailAddresses)+len(x509cr.IPAddresses) != 0 {
+			return
+		}
+
+		csr.Status.Conditions = append(csr.Status.Conditions, certificates.CertificateSigningRequestCondition{
+			Type:    certificates.CertificateApproved,
+			Reason:  "AutoApproved",
+			Message: "Auto approving of all kubelet CSRs is enabled on the controller manager",
+		})
+		csr, err = cc.kubeClient.Certificates().CertificateSigningRequests().UpdateApproval(csr)
```

Comment:
in an error case (like a conflict), won't `csr` end up nil, and `err` end up non-nil?

if so, the `if csr.Status.Certificate == nil && IsCertificateRequestApproved(csr) {` check after calling this func will probably panic. would be better to make this a separate func to avoid closure confusion like this.